include/seastar/http/httpd.hh

```diff
@@ -75,11 +75,12 @@ class connection : public boost::intrusive::list_base_hook<> {
     queue<std::unique_ptr<reply>> _replies { 10 };
     bool _done = false;
     bool _encrypted;
+    int _listener_idx;
```

Comment:
looks great. I'd add on the cover letter that this is needed for the Kafka-Rest compatibility proxy in redpanda as we need to return a contextual information based on HTTP or HTTPS as per the protocol description and need underlying access to the actual listener information.